libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+  }
+
+  /**
+   * Copies a file from one location to another.
+   * By default, destinationPath is overwritten if it already exists.
+   * Behind the scenes it uses `fs.copyFileSync()`.
+   * @param sourcePath - The absolute or relative path to the source file to be copied.
+   * @param destinationPath - The absolute or relative path to the new copy that will be created.
+   */
+  public static copyFile(sourcePath: string, destinationPath: string): void {
+    fsx.copySync(sourcePath, destinationPath);
+  }
+
+  /**
+   * Deletes a file. Can optionally throw if the file doesn't exist.
+   * Behind the scenes it uses `fs.unlinkSync()`.
+   * @param filePath - The absolute or relative path to the file that should be deleted.
+   * @param options - Optional settings that can change the behavior. Type: `IDeleteFileOptions`
+   */
+  public static deleteFile(filePath: string, options?: IDeleteFileOptions): void {
+    options = {
+      throwIfNotExists: false,
+      ...options
+    };
+
+    if (options.throwIfNotExists || FileSystem.exists(filePath)) {
+      fsx.unlinkSync(filePath);
+    }
+  }
+
+  // ===============
+  // LINK OPERATIONS
+  // ===============
+
+  /**
+   * Gets the statistics of a file, folder, or link. Does NOT follow the link to its target.
+   * Behind the scenes it uses `fs.lstatSync()`.
+   * @param path - The absolute or relative path to the file, folder, or link.
+   */
+  public static getLinkStatistics(path: string): fs.Stats {
+    return fsx.lstatSync(path);
+  }
+
+  /**
+   * Creates a symbolic link to a folder (on Windows this is a "directory junction").
+   * Behind the scenes it uses `fs.symlinkSync()`.
+   * @param linkSource - The absolute or relative path to the destination where the link should be created.
+   * @param linkTarget - The absolute or relative path to the target of the link.
+   */
+  public static createSymbolicLinkToFolder(linkSource: string, linkTarget: string): void {
+    // For directories, we use a Windows "junction".  On Unix, this produces a regular symlink.
+    fsx.symlinkSync(linkTarget, linkSource, 'junction');
+  }
+
+  /**
+   * Creates a symbolic link to a file (on Windows this requires elevated permissions).
+   * Behind the scenes it uses `fs.symlinkSync()`.
+   * @param linkSource - The absolute or relative path to the destination where the link should be created.
+   * @param linkTarget - The absolute or relative path to the target of the link.
+   */
+  public static createSymbolicLinkToFile(linkSource: string, linkTarget: string): void {
+    fsx.symlinkSync(linkSource, linkTarget, 'file');
+  }
+
+  /**
+   * Creates a hard link to a file.
+   * Behind the scenes it uses `fs.linkSync()`.
+   * @param linkSource - The absolute or relative path to the destination where the link should be created.
+   * @param linkTarget - The absolute or relative path to the target of the link.
+   */
+  public static createHardLinkToFile(linkSource: string, linkTarget: string): void {
```

Comment:
>createHardLinkToFile [](start = 16, length = 20)

I would call this createHardLink to avoid any confusion about which filesystem objects are supported as the target.